workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -17,7 +26,52 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
   }
 
   public async addAccessPermission(params: AddRemoveAccessPermissionRequest): Promise<PermissionsResponse> {
-    throw new Error('Method not implemented.');
+    if (params.permission.accessLevel !== 'read-only' && params.permission.accessLevel !== 'read-write') {
+      throw new InvalidPermissionError('Access Level must be "read-only" or "read-write".');
+    }
+
+    if (params.permission.identityType !== 'GROUP' && params.permission.identityType !== 'USER') {
+      throw new InvalidPermissionError('IdentityType just be "GROUP" or "USER".');
+    }
+
+    const permissionsEffect: Effect = 'ALLOW';
+    const permissionsAction: Action = params.permission.accessLevel === 'read-only' ? 'READ' : 'UPDATE';
+    const subjectType: IdentityType = params.permission.identityType === 'GROUP' ? 'GROUP' : 'USER';
+
+    const createdPermission: CreateIdentityPermissionsResponse =
+      await this._authorizer.createIdentityPermissions({
+        authenticatedUser: {
+          id: params.authenticatedUserId,
+          roles: params.roles
+        },
+        identityPermissions: [
+          {
+            identityType: subjectType,
+            identityId: params.permission.identity,
+            action: permissionsAction as Action,
```

Comment:
these are already typed above. did the compiler need you to cast them to the same type here?

Reply:
oh, I did that while hunting down a dumb error. Forgot to remove after figuring it out.